sys/linux/vnet.txt

```diff
@@ -508,19 +510,40 @@ ipv6_addr_multicast2 {
 } [packed]
 
 ipv6_addr [
-	rand_addr	array[int8, 16]
+# Few public random addresses
+	rand_addr	ipv6_addr_random
 	empty		ipv6_addr_empty
 # These correspond to LOCAL_IPV6/REMOTE_IPV6/DEV_IPV6 in executor/common_linux.h
 	local		ipv6_addr_t[const[0xaa, int8]]
 	remote		ipv6_addr_t[const[0xbb, int8]]
 	dev		ipv6_addr_t[netdev_addr_id]
 	initdev		ipv6_addr_initdev
+# Some special addresses:
 	loopback	ipv6_addr_loopback
 	ipv4		ipv6_addr_ipv4
 	mcast1		ipv6_addr_multicast1
 	mcast2		ipv6_addr_multicast2
+# Several custom private ranges with 2 addresses each: fc0X::/127.
+# Can be used for custom things within the tested image.
+	private0	ipv6_addr_private[0]
+	private1	ipv6_addr_private[1]
+	private2	ipv6_addr_private[2]
 ] [size[16]]
 
+ipv6_addr_random {
+	a0	const[0x20, int8]
+	a1	const[0x1, int8]
+	a2	array[const[0x0, int8], 13]
+	a3	int8[0:2]
+} [packed, align_4]
```

Comment:
`align_4` isn't needed here and below

Reply:
Doesn't in6_addr have alignment 4?
At least we need to enforce alignment 4 on the whole ipv6_addr union. To do this we need at least 1 member with alignment 4. But adding alignment 4 to just one of them is inconsistent and may break in future if we remove that one for some reason.
Also theoretically ipv6_addr_private can be used on its own somewhere, if we will need to say that that specific address always must be private. Then we will use ipv6_addr_private instead of ipv6_addr, and if we don't have proper alignment, layout may be wrong.

Reply:
Ah, OK. I meant that when `ipv6_addr` is used as a part of packet descriptions, they are `packed` and alignment shouldn't matter. But when it's used in normal structs, yeah, we need to match the alignment of `in6_addr`.